setup.py

```diff
@@ -69,13 +69,9 @@ def build_extension(self, ext):
             ]
             if sys.maxsize > 2**32:
```

Comment:
I think this will be enough to make both visual studio and NMake happy:
```suggestion
            if sys.maxsize > 2**32 and os.environ.get("CMAKE_GENERATOR") != "NMake Makefiles":
```

Reply:
(in addition to the other changes you have here that is)